scripts/release.sh

```diff
@@ -59,43 +60,178 @@ function release_app {
     echo "_____________________________________________"
 }
 
+release_process () {
+
+    # Login in dockerhub with write permissions (repos are public)
+    docker login -u $DOCKER_HUB_USER -p $DOCKER_HUB_PASSWORD
+
+    IMAGE_REPO='ehealthafrica'
+    RELEASE_APPS=( kernel odk couchdb-sync ui producer integration-test )
+
+    echo "_____________________________________________"
+    echo "Release version:   $VERSION"
+    echo "Release revision:  $TRAVIS_COMMIT"
+    echo "Images repository: $IMAGE_REPO"
+    echo "Images:            ${RELEASE_APPS[@]}"
+    echo "_____________________________________________"
+
+    prepare_dependencies
+
+    for APP in "${RELEASE_APPS[@]}"
+    do
+        build_app $APP
+        release_app $APP
+    done
+}
+
+# Usage: increment_version <version> [<position>]
+increment_version () {
+    local v=$1
+    if [ -z $2 ]; then 
+        local rgx='^((?:[0-9]+\.)*)([0-9]+)($)'
+    else 
+        local rgx='^((?:[0-9]+\.){'$(($2-1))'})([0-9]+)(\.|$)'
+        for (( p=`grep -o "\."<<<".$v"|wc -l`; p<$2; p++)); do 
+            v+=.0; done;
+    fi
+    val=`echo -e "$v" | perl -pe 's/^.*'$rgx'.*$/$2/'`
+    TAG_INCREASED_VERSION=$(echo "$v" | perl -pe s/$rgx.*$'/${1}'`printf %0${#val}s $(($val+1))`/)
+}
+
+version_compare () {
+    if [[ $1 == $2 ]]
+    then
+        return 0
+    fi
+    local IFS=.
+    local i ver1=($1) ver2=($2)
+    # fill empty fields in ver1 with zeros
+    for ((i=${#ver1[@]}; i<${#ver2[@]}; i++))
+    do
+        ver1[i]=0
+    done
+    for ((i=0; i<${#ver1[@]}; i++))
+    do
+        if [[ -z ${ver2[i]} ]]
+        then
+            # fill empty fields in ver2 with zeros
+            ver2[i]=0
+        fi
+        if ((10#${ver1[i]} > 10#${ver2[i]}))
+        then
+            return 1
+        fi
+        if ((10#${ver1[i]} < 10#${ver2[i]}))
+        then
+            return 2
+        fi
+    done
+    return 0
+}
+
+function git_branch_commit_and_release() {
+    local BRANCH_OR_TAG_VALUE=$2
+    local REMOTE=origin COMMIT_BRANCH=$TRAVIS_BRANCH
+    if [[ $GITHUB_TOKEN ]]; then
+        REMOTE=https://$GITHUB_TOKEN@github.com/$TRAVIS_REPO_SLUG
+    else
+        echo "Missing environment variable GITHUB_TOKEN=[GitHub Personal Access Token]"
+        exit 1
+    fi
+    version_compare $1 $2
+    COMPARE=$?
+    if [[ ${COMPARE} = 1 ]]
+    then
+        echo "VERSION value" $1 "is greater than" $3 "version" $2
```

Comment:
I think something more needs to happen here. Right now, if I'm on `release-3.7` and the `VERSION` file says `3.7.1`, and then I add commit, the script releases `3.7.0-rc` and sets the `VERSION` file back to `3.7.0` :(